Highlight Admission controller+prep 1.20 (#13838)
DataDog/documentation: content/en/agent/cluster_agent/admission_controller.md

---

## Overview

> The Datadog admission controller is a component of the Datadog Cluster Agent. The main benefit of the admission controller is to simplify the user's application pod configuration. For that, it has two main functionalities:

Reviewer: ```suggestion
The Datadog admission controller is a component of the Datadog Cluster Agent. The main benefit of the admission controller is to simplify your application pod configuration. For that, it has two main functionalities:
```

| `false`          | `admission.datadoghq.com/enabled=true`  | Yes       |
| `false`          | `admission.datadoghq.com/enabled=false` | No        |

#### Order of priority
The Datadog admission controller does not inject the environment variables `DD_VERSION`, `DD_ENV`, and `DD_SERVICE` if they already exist.

> When these environment variables are not set, the admission controller will use standard tags value in the following order (highest first):
> - Labels on the pod
> - Labels on the ownerReference (ReplicaSets, DaemonSets, Deployments...)

Reviewer: ```suggestion
When these environment variables are not set, the admission controller uses standard tags value in the following order (highest first):

- Labels on the pod
- Labels on the `ownerReference` (ReplicaSets, DaemonSets, Deployments...)
```

#### Configure APM and DogstatsD communication mode

> Starting from Datadog Cluster Agent v1.20.0, the Datadog Admission Controller can be configured to inject different mode of communication between the application and Datadog agent.

Reviewer: ```suggestion
Starting from Datadog Cluster Agent v1.20.0, the Datadog Admission Controller can be configured to inject different modes of communication between the application and Datadog agent.
```

This feature can be configured by setting `admission_controller.inject_config.mode` or by defining a pod-specific mode using the `admission.datadoghq.com/config.mode` pod label.

Possible options:
| Mode               | Description                                                                                                       |
|--------------------|-------------------------------------------------------------------------------------------------------------------|
| `hostip` (Default) | Inject the host IP in `DD_AGENT_HOST` environment variable                                                        |

> | `service`          | Inject Datadog's local-service DNS name in `DD_AGENT_HOST` environment variable                                   |

Reviewer 2: perhaps we can add a note that this requires k8s v1.22+

The Datadog admission controller does not inject the environment variables `DD_VERSION`, `DD_ENV`, and `DD_SERVICE` if they already exist.

When these environment variables are not set, the admission controller will use standard tags value in the following order (highest first):
- Labels on the pod
- Labels on the ownerReference (ReplicaSets, DaemonSets, Deployments...)
- `DD_VERSION`, `DD_ENV`, and `DD_SERVICE` set in the container (dockerfile)

#### Configure APM and DogstatsD communication mode
Starting from Datadog Cluster Agent v1.20.0, the Datadog Admission Controller can be configured to inject different mode of communication between the application and Datadog agent.

This feature can be configured by setting `admission_controller.inject_config.mode` or by defining a pod-specific mode using the `admission.datadoghq.com/config.mode` pod label.

Possible options:
| Mode               | Description                                                                     |
|--------------------|---------------------------------------------------------------------------------|
| `hostip` (Default) | Inject the host IP in `DD_AGENT_HOST` environment variable                      |

> | `service`          | Inject Datadog's local-service DNS name in `DD_AGENT_HOST` environment variable |

Reviewer 3: we have a documentation card on our side to document the `network local service` and the new admission controller capability when Agent 7.36 and DCA 1.20 are released.

> | `service`          | Inject Datadog's local-service DNS name in `DD_AGENT_HOST` environment variable |

Reviewer 4: ok great thank you
